sbt-dotty/src/dotty/tools/sbtplugin/DottyPlugin.scala

```diff
@@ -540,10 +564,25 @@ object DottyPlugin extends AutoPlugin {
       dottyVersion,
       fullLoader,
       libraryLoader,
-      Array(dottyLibrary, scalaLibrary),
+      libraryJars,
       compiler,
       all.toArray,
       None)
+  }
+}
+
+private class TopClassLoader(sbtLoader: ClassLoader) extends ClassLoader(null) {
+  private val sharedPrefixes = List(
+    "xsbti.",
+    "org.jline."
+  )
 
+  override protected def loadClass(name: String, resolve: Boolean): Class[_] = {
+    if (sharedPrefixes.exists(name.startsWith(_))) {
+      val c = sbtLoader.loadClass(name)
```

Comment:
I think the comment that was written above this code in CompilerClassLoader is worth preserving (https://github.com/lampepfl/dotty/blob/7b26811cfa827af5ee6d71693020d23827a02e33/sbt-bridge/src/xsbt/CompilerClassLoader.java#L47-L49), might also be worth mentioning that the other overload of loadClass delegates to this one?